### PR TITLE
feat!: migrate kintsugi to session + collator_selection

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

- Removes the `LiquidationVaultFixMigration` as Kintsugi has already been upgraded
- Reorders Kintsugi runtime numbering
- Adds `Session` + `CollatorSelection` pallets
- Adds migration for vanilla `Aura`

### CODE REVIEW

**Comment by sander2** [comment]
Is there a way we can test the correctness of this migration & switch before we upgrade kintsugi?

File: parachain/runtime/kintsugi/src/lib.rs [hunk 1]
```diff
@@ -224,25 +229,205 @@ parameter_types! {
...
+            if frame_system::Pallet::<Runtime>::inc_consumers_without_limit(&account).is_err() {
+                frame_system::Pallet::<Runtime>::inc_providers(&account);
+            }
+        }
+
+        let initial_validators_0 = <CollatorSelection as SessionManager<AccountId>>::new_session_genesis(0)
+            .unwrap_or_else(|| keys.iter().map(|x| x.0.clone()).collect());
+
+        let initial_validators_1 = <CollatorSelection as SessionManager<AccountId>>::new_session_genesis(1)
+            .unwrap_or_else(|| initial_validators_0.clone());
```
Comment (bkchr) [inline parachain/runtime/kintsugi/src/lib.rs:352]:
```suggestion
```

Both are the same anyway, as you just set the invulnerables

**Review by bkchr — COMMENTED** [review]
Generally looking good, but I also didn't look into each detail. I can only recommend you very much that you test this :)

File: parachain/runtime/kintsugi/src/lib.rs [hunk 2]
```diff
@@ -224,25 +229,175 @@ parameter_types! {
...
+    pub const MaxCandidates: u32 = 1000;
+    pub const MinCandidates: u32 = 5;
+    pub const SessionLength: BlockNumber = 6 * HOURS;
+    pub const MaxInvulnerables: u32 = 100;
+    pub const ExecutiveBody: BodyId = BodyId::Executive;
+}
+
+/// We allow root and the Relay Chain council to execute privileged collator selection operations.
+pub type CollatorSelectionUpdateOrigin =
+    EnsureOneOf<EnsureRoot<AccountId>, EnsureXcm<IsMajorityOfBody<ParentLocation, ExecutiveBody>>>;
```
Comment (sander2) [inline parachain/runtime/kintsugi/src/lib.rs:268]: Why allow upgrade through xcm?

Comment (gregdhill) [reply parachain/runtime/kintsugi/src/lib.rs:268]: Same code as on Interlay, allows relay chain governance to set the collators in an emergency.

Comment (sander2) [reply parachain/runtime/kintsugi/src/lib.rs:268]: Not sure this is such a good idea but we can revisit later

**Pull request merged**